File: Week_01/python/80_remove_duplicates_from_sorted_array_ii.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:

        if len(nums) < 3: return len(nums)

        ind, l = 0, len(nums)

        while ind <= l - 3:
            if nums[ind] != nums[ind + 2]:
                ind += 1
            else:
                nums.pop(ind + 2)
                l = len(nums)

        return len(nums)



class Solution:
    def removeDuplicates(self, nums: List[int], k: int) -> int:

        if len(nums) <= k: return len(nums)

        ind, l = 0, len(nums)

        while ind < l - k:
            if nums[ind] != nums[ind + k]:
                ind += 1
            else:
                nums.pop(ind + k)
                l = len(nums)

        return len(nums)

File: Week_01/java/80_remove_duplicates_from_sorted_array_ii.java

class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        if(n <= 2) return n;

        int slow = 2, fast = 2;

        while(fast < n){
            if(nums[slow-2] != nums[fast]) {
                nums[slow] = nums[fast];
                slow++;
            }

            fast++;
        }

        return slow;

    }
}


class Solution {
    public int removeDuplicates(int[] nums, int k) {
        int n = nums.length;
        if(n <= k) return n;

        int slow = k, fast = k;

        while(fast < n){
            if(nums[slow-k] != nums[fast]) {
                nums[slow] = nums[fast];
                slow++;
            }

            fast++;
        }

        return slow;

    }
}
